File: 1160 - Find Words That Can Be Formed by Characters.py
from typing import Counter


class Solution:
    def countCharacters(self, words: list[str], chars: str) -> int:
        d = dict(Counter(chars))
        res = 0
        for word in words:
            temp = dict(Counter(word))
            good = True
            for k in temp.keys():
                if k not in d:
                    good = False
                    break
                elif d[k] < temp[k]:
                    good = False
                    break
                else:
                    continue
            if good:
                res += len(word)
        return res

File: 1160 - Find Words That Can Be Formed by Characters.cpp
#include<unordered_map>
#include<string>
#include<vector>

using namespace std;

class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        unordered_map<char, int> count;
        for (char c : chars) {
            count[c]++;
        }
        int res = 0;
        for (string word : words) {
            unordered_map<char, int> t;
            for (char c : word) {
                t[c]++;
            }
            bool good = true;
            for (auto p = t.begin(); p != t.end(); p++) {
                if (count.find(p->first) == count.end()) {
                    good = false;
                    break;
                } else if (count[p->first] < p->second) {
                    good = false;
                    break;
                }
            }
            if (good) {
                res += word.size();
            }
        }
        return res;
    }
};